encord/objects/internal_helpers.py

```diff
@@ -0,0 +1,470 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Any, Dict, Iterable, List, NoReturn, Optional, Set, Union
+
+from encord.objects.common import (
+    Attribute,
+    ChecklistAttribute,
+    FlatOption,
+    NestableOption,
+    Option,
+    OptionType,
+    PropertyType,
+    RadioAttribute,
+    TextAttribute,
+    _get_option_by_hash,
+)
+from encord.objects.constants import DEFAULT_MANUAL_ANNOTATION
+from encord.objects.utils import (
+    Ranges,
+    _lower_snake_case,
+    ranges_to_list,
+    short_uuid_str,
+)
+
+
+class Answer(ABC):
+    """An internal helper class for the LabelRowClass class. This class is not meant to be used directly."""
+
+    _ontology_attribute: Attribute
+
+    def __init__(self, ontology_attribute: Attribute, track_hash: Optional[str] = None):
+        self._answered = False
+        self._ontology_attribute = ontology_attribute
+        self._track_hash = track_hash or short_uuid_str()
+        self._is_manual_annotation = DEFAULT_MANUAL_ANNOTATION
+        self._should_propagate = False  # DENIS: pass from outside.
+
+    def is_answered(self) -> bool:
+        return self._answered
+
+    def unset(self) -> None:
+        """Remove the value from the answer"""
+        self._answered = False
+        # DENIS: might be better to default everything again for more consistent states!
+
+    @property
+    def is_dynamic(self) -> bool:
+        return self._ontology_attribute.dynamic
+
+    @is_dynamic.setter
+    def is_dynamic(self, value: bool) -> NoReturn:
+        raise RuntimeError("Cannot set the is_dynamic value of the answer.")
+
+    @property
+    def is_manual_annotation(self) -> bool:
```

Comment:
this looks like a normal field :D

Reply:
it becomes it with the setter and getter. Just now no one can do a shady delete (I believe) and the intent is a little clearer to me at least